Source/JavaScriptCore/yarr/YarrPattern.cpp

```diff
@@ -770,18 +814,27 @@ class YarrPatternConstructor {
         else if (!min || (term.type == PatternTerm::Type::ParenthesesSubpattern && m_pattern.m_hasCopiedParenSubexpressions))
             term.quantify(min, max, greedy ? QuantifierType::Greedy : QuantifierType::NonGreedy);
         else {
-            term.quantify(min, min, QuantifierType::FixedCount);
-            m_alternative->m_terms.append(copyTerm(term));
-            // NOTE: this term is interesting from an analysis perspective, in that it can be ignored.....
-            m_alternative->lastTerm().quantify((max == quantifyInfinite) ? max : max - min, greedy ? QuantifierType::Greedy : QuantifierType::NonGreedy);
-            if (m_alternative->lastTerm().type == PatternTerm::Type::ParenthesesSubpattern)
-                m_alternative->lastTerm().parentheses.isCopy = true;
+            if (term.matchDirection() == Forward) {
+                term.quantify(min, min, QuantifierType::FixedCount);
+                m_alternative->m_terms.append(copyTerm(term));
+                // NOTE: this term is interesting from an analysis perspective, in that it can be ignored.....
+                m_alternative->lastTerm().quantify((max == quantifyInfinite) ? max : max - min, greedy ? QuantifierType::Greedy : QuantifierType::NonGreedy);
+                if (m_alternative->lastTerm().type == PatternTerm::Type::ParenthesesSubpattern)
+                    m_alternative->lastTerm().parentheses.isCopy = true;
+            } else {
+                PatternTerm& variableTerm = m_alternative->lastTerm();
```

Comment:
I think this reference is wrong because we are adding a new `m_terms` before using this variableTerm. If Vector is expanded, this reference should point to the wrong memory.

Reply:
Refactored to avoid modifying variableTerm after the append.